### PR TITLE
[app-compat Helm Chart] - Added default topology spread constraint

### PR DESCRIPTION
**Pull request by namannagar89**
JIRA: https://livspaceengg.atlassian.net/browse/DO-18

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the backstage documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: charts/app-compat/templates/deployment.yaml [hunk 1]
```diff
@@ -103,7 +103,27 @@ spec:
         - name: {{ . }}
   {{- end }}
 {{- end}}
-{{- if $.Values.topologySpreadConstraints }}
+{{- $defaultTopologySpreadConstraints := list (dict "maxSkew" 1 "topologyKey" "kubernetes.io/hostname" "whenUnsatisfiable" "ScheduleAnyway" "autoLabelSelector" true) }}
```
Comment (ganarajtejasvi) [inline charts/app-compat/templates/deployment.yaml:106]: define this in _helpers.tpl that is a better place to put it

File: charts/app-compat/templates/deployment.yaml [hunk 2]
```diff
@@ -103,7 +103,27 @@ spec:
...
+{{- if and (not $.Values.topologySpreadConstraints) (eq $.Values.topologySpreadConstraints nil) }}
```
Comment (ganarajtejasvi) [inline charts/app-compat/templates/deployment.yaml:107]: what if the length is 0?

Comment (namannagar89) [reply charts/app-compat/templates/deployment.yaml]: updated this.

File: charts/app-compat/templates/deployment.yaml (hunk 1 above)
Comment (namannagar89) [reply charts/app-compat/templates/deployment.yaml:106]: updated it as per latest discussion

**Review by ganarajtejasvi — APPROVED** [review]

**Pull request merged**